### PR TITLE
[Improve] mybatis-plus updateStrategy improvements

### LINKED ISSUE #3607: [Improve] mybatis-plus updateStrategy improvements
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

StremaPark is using mybatis-plus as an ORM framework. Currently, many entities are annotated with @TableField(updateStrategy = FieldStrategy.IGNORED). As the number of entity classes grows, this has caused trouble for the code's readability and maintainability. This issue can be addressed by using mybatis-plus's global configuration to remove the updateStrategy annotation from all entity classes. However, this may result in some of the original data update operations failing, requiring us to make adjustments. The general approach to solving this is to retrieve and populate the entity from the database before updating. Do you have a better solution? Let's discuss! 😉
image

mybatis global update strategy:

image

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by VampireAchao**
## What changes were proposed in this pull request

Issue Number: close #3607 

## Brief change log

- [x] Updated the global `updateStrategy` setting in MyBatis-Plus.
- [x] Removed the `@TableField(updateStrategy = FieldStrategy.IGNORED)` annotations across all entity classes.
- [x] Enumerated and verified all affected update operations to ensure integrity and prevent issues.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*no* 

This change is already covered by existing tests, such as *(please describe tests)*.

*no*

This change added tests and can be verified as follows:

no

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): yes
 - add cn.hutool:hutool-core:5.8.27

### CODE REVIEW

**Comment by wolfboys** [comment]
good job👍

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java [hunk 1]
```diff
@@ -116,10 +116,9 @@ public RestResponse toggleToken(Long tokenId) {
             ? AccessToken.STATUS_DISABLE
             : AccessToken.STATUS_ENABLE;
 
-    AccessToken updateObj = new AccessToken();
-    updateObj.setStatus(status);
-    updateObj.setId(tokenId);
-    return RestResponse.success(this.updateById(updateObj));
+    tokenInfo.setStatus(status);
+    tokenInfo.setId(tokenId);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java]: We don't need to set id

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+class ExternalLinkServiceTest extends SpringUnitTestBase {
+  @Autowired private ExternalLinkService externalLinkService;
+
+  @AfterEach
+  void cleanTestRecordsInDatabase() {
+    externalLinkService.remove(new QueryWrapper<>());
+  }
+
+  @Test
+  void testUpdate() {
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java]: void testUpdateLinkService()

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    link.setBadgeName("badgeName");
+    link.setLinkUrl("linkUrl");
+    externalLinkService.save(link);
+    link.setBadgeLabel("updatedBadgeLabel");
+    link.setBadgeColor("updatedBadgeColor");
+    link.setBadgeName("updatedBadgeName");
+    link.setLinkUrl(null);
+    externalLinkService.update(link);
+    link = externalLinkService.getById(link.getId());
+    assertThat(link.getBadgeLabel()).isEqualTo("updatedBadgeLabel");
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java]: Can I add some space here? I feel too crowded

**Review by caicancai — CHANGES_REQUESTED** [review]

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java (hunk 3 above)
Comment (VampireAchao) [reply streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/ExternalLinkServiceTest.java]: okay，thanks for suggestion 🧸

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/application/ApplicationActionServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  @Autowired private ApplicationActionService applicationActionService;
+
+  @AfterEach
+  void cleanTestRecordsInDatabase() {
+    applicationActionService.remove(new QueryWrapper<>());
+  }
+
+  @Test
+  void testUpdateApplicationActionServiceById() {
+    Application app = new Application();
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/application/ApplicationActionServiceTest.java]: I'm not sure what the significance of this test is. In my opinion, it's just an interface similar to updating dto. Do we need to write a test class specifically for this?

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/alert/AlertConfigServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+class AlertConfigServiceTest extends SpringUnitTestBase {
+  @Autowired private AlertConfigService alertConfigService;
+
+  @AfterEach
+  void cleanTestRecordsInDatabase() {
+    alertConfigService.remove(new QueryWrapper<>());
+  }
+
+  @Test
+  void testUpdateAlertConfigServiceById() {
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/alert/AlertConfigServiceTest.java]: above all

**Review by caicancai — CHANGES_REQUESTED** [review]

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/application/ApplicationActionServiceTest.java (hunk 4 above)
Comment (VampireAchao) [reply streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/application/ApplicationActionServiceTest.java]: I will remove theses test after them passed.

**Comment by VampireAchao** [comment]
@caicancai @RocMarshal @wolfboys this pr is ready for review

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/utils/BeanUtil.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.apache.streampark.console.core.utils;
+
+import cn.hutool.core.bean.copier.CopyOptions;
+import cn.hutool.core.lang.func.Func1;
+
+/** Util class for bean */
+public class BeanUtil {
+
+  /**
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/utils/BeanUtil.java:26]: Can this annotation be improved or the format be better looking?

File: streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/SavePointServiceTest.java [hunk 7]
```diff
@@ -115,6 +115,7 @@ void testGetSavepointFromAppCfgIfStreamParkOrSQLJob() {
     appCfg.setFormat(ConfigFileTypeEnum.PROPERTIES.getValue());
     configService.save(appCfg);
     assertThat(savePointServiceImpl.getSavepointFromConfig(app)).isNull();
+    appCfg = configService.getById(appCfgId);
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/SavePointServiceTest.java]: appConfigId

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java [hunk 8]
```diff
@@ -116,14 +117,22 @@ public RestResponse toggleToken(Long tokenId) {
...
+    return RestResponse.success(this.updateById(tokenInfo));
   }
 
   @Override
   public AccessToken getByUserId(Long userId) {
     return baseMapper.selectByUserId(userId);
   }
+
+  @Override
+  public boolean updateById(AccessToken entity) {
```
Comment (caicancai) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java:130]: I noticed that there are many similar writing methods. Can it be optimized through templates or options?

**Review by caicancai — CHANGES_REQUESTED** [review]

**Comment by caicancai** [comment]
@VampireAchao Thank you for your contribution. I have left some comments. You can check and modify them. They are all minor issues.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/utils/BeanUtil.java (hunk 6 above)
Comment (VampireAchao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/utils/BeanUtil.java:26]: This is the default formatting style used by Spotless.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java (hunk 8 above)
Comment (VampireAchao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/AccessTokenServiceImpl.java:130]: I do not understand

**Review by wolfboys — APPROVED** [review]
LGTM

**Review by caicancai — APPROVED** [review]
LGTM

**Pull request merged**